mvc/controllers/areas.rst

```diff
@@ -1,8 +1,132 @@
+
+.. code-block:: c#
+  :emphasize-lines: 4
+
+  ...
+  namespace MyStore.Areas.Products.Controllers
+  {
+      [Area("Products")]
+      public class HomeController : Controller
+      {
+          // GET: /<controller>/
+          public IActionResult Index()
+          {
+              return View();
+          }
+      }
+  }
+
+The final step is to set up a route definition that works with your newly created areas. The :doc:`routing` article goes into detail about how to create route definitions, including using conventional routes versus attribute routes. In this example, we'll use a conventional route. To do so, simply open the ``startup.cs`` file and modify it by adding the highlighted route definition below.
+
+.. code-block:: c#
+  :emphasize-lines: 4-6
+
+  ...
+  app.UseMvc(routes =>
+  {
+      routes.MapRoute(name: "areaRoute",
+              template: "{area:exists}/{controller}/{action}",
+              defaults: new { controller = "Home", action = "Index" });
+
+      routes.MapRoute(
+          name: "default",
+          template: "{controller=Home}/{action=Index}");
+  });
+
+Now, when the user enters http://<yourApp>/products, the Index action method of the ``HomeController`` in the ``Products`` area will be invoked and its view will be sent to the client for rendering.
+
+Linking between areas
+---------------------
+
+Once you've set up your areas, you'll need to know how to link between areas. By way of example, take a look at the following hierarchy.
+
+- Project name
+
+  - Areas
+
+    - Products
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+  - Controllers
+
+    - HomeController
+
+  - Views
+
+    - Home
+
+      - Index
+
+As you can see, this example has two ``HomeController`` controllers; one that's defined in an area (``Products/Controllers/HomeController``) and one that's defined outside an area (``Controllers/HomeController``). Let's say you want to put a link on each page that links back to the other page.
+
+Linking from outside to inside
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Linking from outside to inside simply means to link  from a place that's outside of any area to an action that's inside an area. For example, linking from ``Controllers/HomeController`` to ``Products/Controllers/HomeController``. To do this, you could simply use the ``Html.ActionLink`` method and specify the area as an anonymous object.
+
+.. code-block:: c#
+
+  @Html.ActionLink("See Products Home Page", "Index", "Home", new { area = "Products" }, null)
+
+Linking from inside to outside
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Linking from inside to outside simply means to link from a place that's inside an area to an action that's outside an area. As you can probably guess, you simply specify an area of "".
+
+.. code-block:: c#
+
+  @Html.ActionLink("Go to Home Page", "Index", "Home", new { area = "" }, null)
+
```

Comment:
I feel like this whole section would be clearer if it just said: You disambiguate controllers by including the area in the route values. Use an empty string for controllers that aren't within an area.  And then include the two code examples.